src/modules/hub.bicep

```diff
@@ -60,40 +72,273 @@ resource defaultTelemetry 'Microsoft.Resources/deployments@2022-09-01' = if (ena
+resource dataContainer 'Microsoft.Storage/storageAccounts/blobServices/containers@2021-06-01' =  {
+  parent: blobService
+  name: 'ingestion'
+  properties: {
+    publicAccess: 'None'
+    metadata: {}
+  }
+}
+
+resource exportContainer 'Microsoft.Storage/storageAccounts/blobServices/containers@2021-06-01' =  {
+  parent: blobService
+  name: 'ms-cm-exports'
+  properties: {
+    publicAccess: 'None'
+    metadata: {}
+  }
+}
+
+resource uploadSettingsJson 'Microsoft.Resources/deploymentScripts@2020-10-01' = {
+  name: 'updateSettingsJson'
+  kind: 'AzurePowerShell'
+  location: location
+  dependsOn: [
+    configContainer
+  ]
+  properties: {
+    azPowerShellVersion: '8.0'
+    retentionInterval: 'PT1H'
+    environmentVariables: [
+      {
+        name: 'exportScopes'
+        value: join(exportScopes, '|')
+      }
+      {
+        name: 'storageAccountKey'
+        value: storageAccount.listKeys().keys[0].value
+      }
+      {
+        name: 'storageAccountName'
+        value: storageAccountName
+      }
+      {
+        name: 'containerName'
+        value: 'config'
+      }
+    ]
+    scriptContent: loadTextContent('./scripts/Copy-FileToAzureBlob.ps1')
+  }
+}
+
+
+// Key Vault for storing secrets
+
+module keyVault 'Microsoft.KeyVault/vaults/deploy.bicep' = {
+  name: 'keyVault'
+  dependsOn: [
+    dataFactory
+  ]
   params: {
-    name: storageAccountName
+    name: keyVaultName
     location: location
-    storageAccountSku: storageSku
     tags: resourceTags
+    enablePurgeProtection: false
+    accessPolicies: [
+      {
+        objectId: dataFactory.outputs.systemAssignedPrincipalId
+        tenantId: subscription().tenantId
+        permissions: {
+          secrets: [
+            'get'
+          ]
+        }
+      }
+    ]
+  }
+}
+
+module keyVaultSecret 'secrets/exports.bicep' = {
+  name: 'keyVaultSecret'
+  dependsOn: [
+    storageAccount
+    keyVault
+  ]
+  params: {
+    keyVaultName: keyVaultName
+    storageAccountName: storageAccountName
   }
 }
 
+// Azure Data Factory and Pipelines
 module dataFactory 'Microsoft.DataFactory/factories/deploy.bicep' = {
   name: 'dataFactory'
   params: {
     name: dataFactoryName
     location: location
     tags: resourceTags
+    systemAssignedIdentity: true
   }
 }
 
-/**
- * Outputs
- */
+module linkedService_keyVault 'linkedservices/keyvault.bicep' = {
+  name: 'linkedService_keyVault'
+  dependsOn: [
+    dataFactory
+    keyVault
+  ]
+  params: {
+    dataFactoryName: dataFactoryName
+    keyVaultName: keyVaultName
+  }
+}
+
+module linkedService_storageAccount 'linkedservices/ms_cm_exports.bicep' = {
+  name: 'linkedService_storageAccount'
+  dependsOn: [
+    dataFactory
+    keyVault
+    linkedService_keyVault
+    keyVaultSecret
+  ]
+  params: {
+    dataFactoryName: dataFactoryName
+    keyVaultName: keyVaultName
+    storageAccountName: storageAccountName
+  }
+}
+
+module dataset_exports 'datasets/ms_cm_exports.bicep' = {
+  name: 'dataset_exports'
```

Comment:
Use `ms-cm-exports` so it's unique going forward.

Reply:
Done